core/src/smartcontracts/isi/query.rs

```diff
@@ -384,7 +384,7 @@ mod tests {
             .sign(ALICE_KEYS.clone())?;
         let wrong_hash = unapplied_tx.hash();
         let not_found = FindTransactionByHash::new(wrong_hash).execute(&wsv);
-        assert!(matches!(not_found, Err(_)));
+        assert!(not_found.is_err());
```

Comment:
Another option is to set the exact `Err` type. Given that these errors are user facing, I'd hard-code the error type into the `matches!` invocation.